spec/controllers/checkin_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+include FactoryBot::Syntax::Methods
+
+describe CheckinController do
+  before do
+    # set precondition: sessions, datetime, checkin records
+    @stu = create :student
+    @expected_time = DateTime.parse("2022-03-08T12:00:00-08:00")
+    allow(DateTime).to receive(:now).and_return @expected_time
+    controller.session[:current_user_id] = @stu.id
+  end
+
+  describe "POST create" do
+    before do
+      @n_checkin_before = Checkin.all.size
+      post :create
+      @new_checkin = Checkin.order(id: :desc).limit(1)[0]
```

Comment:
just a syntax thing, but I think you can achieve this using `.first` instead of `.limit(1)[0]`